File: reporting_frontend/src/client/app/shared/models/feedbacks/rating_feedback.ts
import {RatingMechanism} from '../mechanisms/rating_mechanism';


export class RatingFeedback {
    rating:number;
    mechanismId:number;
    mechanism:RatingMechanism;

    constructor(rating:number, mechanismId:number) {
        this.rating = rating;
        this.mechanismId = mechanismId;
    }
}

File: android_library/feedbacklibrary/src/main/java/ch/uzh/supersede/feedbacklibrary/feedbacks/RatingFeedback.java
package ch.uzh.supersede.feedbacklibrary.feedbacks;

import com.google.gson.annotations.Expose;

import java.io.Serializable;

import ch.uzh.supersede.feedbacklibrary.models.RatingMechanism;

public class RatingFeedback implements Serializable {
    @Expose
    private long mechanismId;
    @Expose
    private long rating;

    public RatingFeedback(RatingMechanism ratingMechanism) {
        setMechanismId(ratingMechanism.getId());
        setRating((long) ratingMechanism.getInputRating());
    }

    public long getMechanismId() {
        return mechanismId;
    }

    public void setMechanismId(long mechanismId) {
        this.mechanismId = mechanismId;
    }

    public long getRating() {
        return rating;
    }

    public void setRating(long rating) {
        this.rating = rating;
    }
}
